### PR TITLE
[do-not-merge-before-28] Enable updates for ScalableTarget and ScalingPolicy 

### PR DESCRIPTION
**Pull request by mbaijal**
Issue #, if available:
https://github.com/aws-controllers-k8s/community/issues/867

Description of changes:
1. Enable updates based on changes to code generator made in this PR - https://github.com/aws-controllers-k8s/code-generator/pull/166
2. Add an integration test to check for updates
3. Update Runtime to v0.12.0
4. Allows use of adopted resource (test in next PR)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,8 +3,8 @@ module github.com/aws-controllers-k8s/applicationautoscaling-controller
 go 1.14
 
 require (
-	github.com/aws-controllers-k8s/runtime v0.2.1
-	github.com/aws/aws-sdk-go v1.37.4
+	github.com/aws-controllers-k8s/runtime v0.12.0
```
Comment (A-Hilaly) [inline go.mod:6]: @mbaijal I would recommend to do the runtime update in a separate PR, since will require regenerating all the apis and tuning/re-running all the unit tests

Comment (RedbackThomson) [reply go.mod:6]: Agree. It's quite difficult to see in this review what is simply updating to 0.12.0 and what is custom code. Especially considering we are coming from 0.2.1 - there are a lot of rebasing changes.

Comment (mbaijal) [reply go.mod:6]: I had to update the runtime here since I was using the latest code generator changes for this PR which includes late initializers and without this version of runtime things don't work. 

I could however have a separate PR for runtime update which gets merged before this one.

Comment (RedbackThomson) [reply go.mod:6]: Yes please. A separate PR for runtime from v0.2.1 to v0.12.0 will greatly reduce subsequent PRs (like this one)

**Comment by mbaijal** [comment]
Will create a new pull request since runtime update is in a separate PR now and there have been other changes since.
